blog/2024-05-25-bring-slack.md

```diff
@@ -0,0 +1,266 @@
+![Creating from scratch](assets/2024-05-25-bring-slack/from-scratch.png)
+3. For the sake of our tutorial, we will be building a bot that requires the following permissions:
+  - `app_mentions:read`
+  - `chat:write`
+  - `chat:write.public`
+to add these head over to the `OAuth & Permissions` section and add those permissions.
+![Adding permissions](assets/2024-05-25-bring-slack/permissions.png)
+4. Install the app to your workspace and copy the bot token.
+![Install app](assets/2024-05-25-bring-slack/install-to-workspace.png)
+
+Once its installed you can copy the bot token and let's head back over to our Wing app.
+
+## Configuring Slack Bot Token Secret
+
+Now that we have our bot token, we can add it to our application by running the `wing secrets` command and pasting the token when prompted:
+
+```bash
+❯ wing secrets
+1 secret(s) found
+
+? Enter the secret value for slack-bot-token: [hidden]
+```
+
+Now that our bot token is stored our application is ready to run!
+
+## Running the App Locally
+
+To run the app locally we can use the Wing Console, which simulates the cloud environment on your local machine. To start the console run:
+
+```bash
+wing it
+```
+This will open a browser window showing the Wing Console. You should see something similar to this:
+
+![Wing Console](assets/2024-05-25-bring-slack/wing-console.png)
+
+Now in order to see the Slack bot in action, let's add some more code to our `main.w` file. We will add a function that will make a new file each time it is called.
+
+The following code can be appended to the `main.w` file:
+
+```js
+let counter = new cloud.Counter();
+
+new cloud.Function(inflight () => {
+  let i = counter.inc();
+  inbox.put("file-{i}.txt", "Hello, Slack!");
+}) as "Add File";
+```
+
+Once you save the file, the Wing Console will hot reload and you should now see a function resource we can play with that looks like this:
+
+![Add File Function](assets/2024-05-25-bring-slack/add-file-function.png)
+
+So now we can click on the `Add File` function and interact with it in the right side panel. Go ahead and invoke the function a few times.
+
+![Invoke Function](assets/2024-05-25-bring-slack/invoke-function.png)
+
+And BAM!! You should now be seeing messages in your slack channel every time you invoke the function!
+
+![Slack Messages](assets/2024-05-25-bring-slack/slack-messages.png)
+
+
+## Enabling Events
+
+One thing we will notice is the Slack application is supposed to support the ability to list the files in the inbox. This is done by mentioning the Slack app and saying `list inbox`. 
+
+If you try this now you will see absolutely nothing happens :) —— this is because we need to enable events in our slack app.
+
+To do this head over to the `Event Subscriptions` section in the slack API dashboard and enable events. You will need to provide a URL for the slack API to send events to. Luckily Wing makes providing this URL easy with builtin support for tunneling.
```

Comment:
I'm just wondering if it will be confusing to the reader when you say to head over to the `Event Subscriptions` section, but they can't because they need a URL first. Maybe something like "Before we head to the ..."

Because you then direct them to head there a couple of paragraphs later.